docs/figure_block.jl

```diff
@@ -5,18 +5,129 @@ Documenter.Selectors.order(::Type{FigureBlocks})  = 8.0 # like @example
+        factor = round.(Int, size(img) ./ size_px)
+        ntrim = 3
+        img = @view ImageTransformations.restrict(img)[ntrim:end-ntrim,ntrim:end-ntrim]
+        # img = @view ImageTransformations.restrict(img)[ntrim:end-ntrim,ntrim:end-ntrim]
+        final_size_px = Tuple(round.(Int, size(img) ./ factor))
+        io = IOBuffer()
+        FileIO.save(FileIO.Stream{FileIO.format"PNG"}(Makie.raw_io(io)), img)
+        push!(vec, Png(take!(io), final_size_px, id))
+        return
+    end
+
+    struct FileInfo
+        filename::String
+        id::String
+        size_px::Tuple{Int,Int}
+    end
+    struct OverviewSection
+        d::Dict{PageInfo,Vector{FileInfo}}
+    end
+
+    function OverviewSection(page::String)
+        r = Regex("/$page/")
+
+        filtered = filter(pairs(FIGURES)) do (pageinfo, pngs)
+            match(r, pageinfo.path) !== nothing
+        end
+
+        fileinfo_dict = Dict{PageInfo,Vector{FileInfo}}()
+        for (pageinfo, pngs) in pairs(filtered)
+            fileinfos = map(pngs) do png
+                filename = "$(string(hash(png.bytes), base = 62)).png"
+                open(filename, "w") do _io
+                    write(_io, png.bytes)
+                end
+                return FileInfo(filename, png.id, png.size_px)
+            end
+            fileinfo_dict[pageinfo] = fileinfos
+        end
+
+        OverviewSection(fileinfo_dict)
+    end
+
+    function Base.show(io::IO, ::MIME"text/markdown", o::OverviewSection)
+        pages = sort(collect(keys(o.d)), by = x -> x.path)
+        for page in pages
+            fileinfos = o.d[page]
+            pagename, _ = splitext(basename(page.path))
+            println(io, "### [$(page.title)](@id overview_$(pagename))")
+            println(io)
+            println(io, """<div :class="\$style.container">""")
+            for fileinfo in fileinfos
+                println(io, """
+                <a href="./$pagename.html#example-$(fileinfo.id)">
+                    <img src=\"./$(fileinfo.filename)\" />
+                </a>
+                """)
+            end
+            println(io, "</div>")
+            println(io)
+        end
+
+        println(io, """
+        <style module>
+            .container {
+                display: grid;
+                grid-template-columns: repeat(2, 1fr);
+                grid-gap: 1.5em;
+                padding: 2em 0;
+            }
+
+            @media (min-width: 640px) {
+                .container {
+                    grid-template-columns: repeat(3, 1fr);
+                }
+            }
+
+            @media (min-width: 960px) {
+                .container {
+                    grid-template-columns: repeat(4, 1fr);
+                }
+            }
+        </style>
+        """)
+    end
 end
 
 
 
 function Documenter.Selectors.runner(::Type{FigureBlocks}, node, page, doc)
+    title = first(Iterators.filter(page.elements) do el
+        el isa Markdown.Header{1}
+    end).text[]
```

Comment:
This will only work for the first text element - I suggest mdflatten on the node.

Reply:
what do you mean by first text element, could there be more? I'm not sure why it's even a vector

Reply:
If there's any code that's one element, then text, then code again, etc.  For example,
```
# `qqnorm` and `qqplot`
```
expands to
```julia
julia> convert(MarkdownAST.Node, Markdown.parse("# `qqnorm` and `qqplot`"))
@ast MarkdownAST.Document() do
  MarkdownAST.Heading(1) do
    MarkdownAST.Code("qqnorm")
    MarkdownAST.Text(" and ")
    MarkdownAST.Code("qqplot")
  end
end
```

Reply:
and 
```julia-repl
julia> mda = convert(MarkdownAST.Node, Markdown.parse("# `qqnorm` and `qqplot`"))
@ast MarkdownAST.Document() do
  MarkdownAST.Heading(1) do
    MarkdownAST.Code("qqnorm")
    MarkdownAST.Text(" and ")
    MarkdownAST.Code("qqplot")
  end
end


julia> Documenter.MDFlatten.mdflatten(mda)
"qqnorm and qqplot\n\n"
```